File: app/src/main/java/com/sprotte/geolocator/demo/java/NotificationWorker.java
package com.sprotte.geolocator.demo.java;

import android.content.Context;

import androidx.annotation.NonNull;

import com.sprotte.geolocator.demo.misc.UtilsKt;
import com.sprotte.geolocator.geofencer.models.GeoFenceUpdateModule;
import com.sprotte.geolocator.geofencer.models.Geofence;

import timber.log.Timber;

public class NotificationWorker extends GeoFenceUpdateModule {

    private Context context;
    NotificationWorker(Context context){
        super(context);
        this.context = context;

    }

    @Override
    public void onGeofence(@NonNull Geofence geofence) {
        Timber.d("NotificationWorker JAVA : onGeofence ");
        UtilsKt.sendNotification(context, geofence.getTitle(), geofence.getMessage());
    }
}

File: app/src/main/java/com/sprotte/geolocator/demo/kotlin/NotificationWorker.kt
package com.sprotte.geolocator.demo.kotlin

import android.content.Context
import com.sprotte.geolocator.demo.misc.sendNotification
import com.sprotte.geolocator.geofencer.models.GeoFenceUpdateModule
import com.sprotte.geolocator.geofencer.models.Geofence
import timber.log.Timber

class NotificationWorker (private val context: Context): GeoFenceUpdateModule(context){
    override fun onGeofence(geofence: Geofence) {
        Timber.d("onGeofence $geofence")
        sendNotification(
            context = context,
            title = geofence.title,
            message = geofence.message
        )
    }

}